chia/util/streamable.py

```diff
@@ -700,3 +714,15 @@ class UInt32Range(Streamable):
 class UInt64Range(Streamable):
     start: uint64 = uint64(0)
     stop: uint64 = uint64.MAXIMUM
+
+
+_T_Enum = TypeVar("_T_Enum", bound=EnumMeta)
+
+
+def streamable_enum(proxy: type[Any]) -> Callable[[_T_Enum], _T_Enum]:
```

Comment:
this, or can we require the proxy itself be streamable?

```suggestion
def streamable_enum(proxy: type[object]) -> Callable[[_T_Enum], _T_Enum]:
```

maybe not in this pr, but maybe we need a type that is like a union of the rust streamableness and `Streamable`.  or...